packages/transport/test/modules/relationships/decomposition.test.ts

```diff
@@ -118,3 +118,42 @@ describe("Data cleanup after relationship decomposition", function () {
         );
     });
 });
+
+describe("Relationship decomposition due to Identity deletion", function () {
+    let connection: IDatabaseConnection;
+    let transport: Transport;
+
+    let sender: AccountController;
+    let recipient: AccountController;
+
+    beforeAll(async function () {
+        connection = await TestUtil.createDatabaseConnection();
+        transport = TestUtil.createTransport(connection);
+
+        await transport.init();
+
+        const accounts = await TestUtil.provideAccounts(transport, 2);
+        sender = accounts[0];
+        recipient = accounts[1];
+
+        await TestUtil.addRelationship(sender, recipient);
+    });
+
+    afterAll(async function () {
+        await sender.close();
+        await recipient.close();
+        await connection.close();
+    });
+
+    test("deletion of a previously active Relationship should be proposed after the peer is deleted", async function () {
+        const activeRelationship = await sender.relationships.getActiveRelationshipToIdentity(recipient.identity.address);
+        expect(activeRelationship?.status).toBe(RelationshipStatus.Active);
+
+        await recipient.identityDeletionProcess.initiateIdentityDeletionProcess(0);
+        await TestUtil.runDeletionJob();
+
+        await sender.syncEverything();
+        const deletionProposedRelationship = await sender.relationships.getRelationshipToIdentity(recipient.identity.address);
+        expect(deletionProposedRelationship?.status).toBe(RelationshipStatus.DeletionProposed);
+    });
```

Comment:
```suggestion
    test("RelationshipStatus of a previously active Relationship should be set to DeletionProposed after the peer is deleted", async function () {
        const activeRelationship = await sender.relationships.getActiveRelationshipToIdentity(recipient.identity.address);
        expect(activeRelationship!.status).toBe(RelationshipStatus.Active);

        await recipient.identityDeletionProcess.initiateIdentityDeletionProcess(0);
        await TestUtil.runDeletionJob();

        await sender.syncEverything();
        const deletionProposedRelationship = await sender.relationships.getRelationshipToIdentity(recipient.identity.address);
        expect(deletionProposedRelationship!.status).toBe(RelationshipStatus.DeletionProposed);
    });
```

From my point of view the test name is a bit unambiguous about whether something is actually proposed to the Identity when in truth only the status is changed.

Reply:
@britsta Please mind the (hidden) `!`.

Reply:
Oops, I didn't see them at first. :D

Why is it necessary to insert two `!` at these locations?

Reply:
Technically it isn't required, the `?` serve the same purpose. However, in tests we do expect the objects to be defined and if not we want an error to be thrown immediately. So `!` are favored. In productive code, on the other hand, `?` should be used instead, as we don't want it to break on an error.

Reply:
actually in that case it doesn't matter if ! or ? is used as toBe would fail when it's undefined.

Reply:
As I understand it, it's more of a high level thought behind it than actually having an impact on the failure or success of the test.